[Terence Wong] iP

TerenceWongsExtravaganza opened the pull request:
Bean
> “Gedagedigedagedao.” – Nugget 

Bean frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

[x] Managing tasks
[x]Managing deadlines 
[ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TerenceWongsExtravaganza committed:
completed level 1

TerenceWongsExtravaganza committed:
completed level 2

TerenceWongsExtravaganza committed:
completed level 3

TerenceWongsExtravaganza committed:
completed level 4

TerenceWongsExtravaganza committed:
ui tests

TerenceWongsExtravaganza committed:
completed level 5

TerenceWongsExtravaganza committed:
completed level 6

TerenceWongsExtravaganza committed:
Add feature to read from tasks.txt

TerenceWongsExtravaganza committed:
Add feature to write onto tasks.txt

TerenceWongsExtravaganza committed:
Add Storage class for future use

TerenceWongsExtravaganza committed:
Refactor code into different methods

TerenceWongsExtravaganza committed:
Change Deadline and Event class to separate out dates

TerenceWongsExtravaganza committed:
Add feature to format dates

TerenceWongsExtravaganza committed:
Made code more OOP by splitting in classes

TerenceWongsExtravaganza committed:
Refactor code into packages

TerenceWongsExtravaganza committed:
Merge branch 'add-gradle-support'

TerenceWongsExtravaganza committed:
Add gradle

TerenceWongsExtravaganza committed:
Add test for Deadline and Todo class

TerenceWongsExtravaganza committed:
Update build.gradle for Jar build

TerenceWongsExtravaganza committed:
Add JavaDocs to Parser and UI class

TerenceWongsExtravaganza committed:
Add javadocs to public method and classes

TerenceWongsExtravaganza committed:
Fix coding standard for certain classes

TerenceWongsExtravaganza committed:
Fix parser bug

TerenceWongsExtravaganza committed:
Fix to coding standard

TerenceWongsExtravaganza committed:
Add message for unmarking

TerenceWongsExtravaganza committed:
Add find function

TerenceWongsExtravaganza committed:
Revert "Add find function"

This reverts commit 318f54bc3c09213c1a6667c159ef019d1f5ec833.

TerenceWongsExtravaganza committed:
Add find feature

TerenceWongsExtravaganza committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/main/Parser.java
#	src/main/java/main/TaskList.java

TerenceWongsExtravaganza committed:
Merge branch 'branch-CodingStandard'

# Conflicts:
#	src/main/java/main/Parser.java
#	src/main/java/main/TaskList.java
#	src/main/java/main/Ui.java

TerenceWongsExtravaganza committed:
Add checkstyles

O-Darren commented on src/test/java/task/DeadlineTest.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Test
+    public void emptyTodo() throws Exception {
```
Perhaps this could be named testEmptyDeadline. Methods are normally named as verbs, and this method tests for an empty Deadline rather than an empty Todo.

O-Darren commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,172 @@
+     * Returns size of taskList to be used for loops or other operations
+     * @return size of taskList
+     */
+    public int size() {
```
Perhaps this could be named getSize(). Methods are usually named after verbs.

O-Darren commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like that the boolean values sound like boolean values.

O-Darren commented on src/test/java/task/TodoTest.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Test
+    public void emptyTodo() throws Exception {
```
Perhaps this could be named testEmptyTodo. Methods are normally named after verbs.

O-Darren reviewed (commented):
LGTM! Naming is generally well done, with nouns for class names and verbs for method names.

TerenceWongsExtravaganza committed:
Add JavaFX gui and modularise commands

TerenceWongsExtravaganza committed:
Merge branch 'branch-Level-10'

jolynleow2301 commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     */
+    public static Command handleBye() {
+        return new MiscCommand("________________________________\n"
+                + "Bye. Hope to see you again soon!\n"
```
Good adherence to Coding Standards line length

jolynleow2301 commented on src/main/java/main/Storage.java:
```diff
@@ -0,0 +1,91 @@
+                        || splitResponse[1].equals("deadline")) {
+                    Task currentTask = null;
+                    String[] taskDetails;
+                    try {
```
Adherence to Coding Standards (Egyptian style)

jolynleow2301 commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+    /**
+     * Creates TaskList object with the given taskList by the Storage class
+     * @param taskList ArrayList of type Task to be be used
+     */
```
Header comment should be inserted just before the method

jolynleow2301 commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+    /**
+     * Displays the list of tasks in taskList
+     */
+    public String list() {
```
Perhaps this could be named getList(), as that is the function of the list() here and it will adhere to Coding Standard

jolynleow2301 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+        }
+    }
+
+    @Override
```
Will be good to include the header comments for this method as well

jolynleow2301 reviewed (commented):
Looks good. Just some nits to fix.

TerenceWongsExtravaganza committed:
Fix checkstyle errors

TerenceWongsExtravaganza committed:
Implement assertion to multiple classes

Most adds assertion to different command class to make sure
commands are valid

Assertion added to parser class to prevent invalid inputs

TerenceWongsExtravaganza committed:
Refactor code in Storage and TaskList class for better
readability

TerenceWongsExtravaganza committed:
Merge pull request #1 from TerenceWongsExtravaganza/branch-A-Assertions

Implement assertion to multiple classes

TerenceWongsExtravaganza committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/main/TaskList.java

TerenceWongsExtravaganza committed:
Enhance code quality of several command class

Add description of what method does and restructured code for
better readability

Deleted Some redundant code

TerenceWongsExtravaganza committed:
Merge pull request #2 from TerenceWongsExtravaganza/branch-A-CodeQuality

Branch a code quality

TerenceWongsExtravaganza committed:
Add .yml file to set up GitHub Actions

TerenceWongsExtravaganza committed:
Remove workflow files

TerenceWongsExtravaganza committed:
Add C-Update to IP

Allows updating details of different task type, such as
deadline for Deadline task and start/end date
for Event task

TerenceWongsExtravaganza committed:
Change personality of Bean

Made Personality of Bean more unique and fun

TerenceWongsExtravaganza committed:
Merge pull request #4 from TerenceWongsExtravaganza/branch-A-Personality

Change personality of Bean

TerenceWongsExtravaganza committed:
Add user guide and change UI elements

Change title of window and adds custom images

TerenceWongsExtravaganza committed:
Fix storage bug when running JAR file

Change README
